### PR TITLE
WA-136: Add support for scanning ERC67 schemes.

### PR DESCRIPTION
**Pull request by fmrsabino**
Changes proposed in this pull request:
- Some fields with address input only supported raw addresses. This adds support for ERC67 in those fields.

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/pm/gnosis/heimdall/ui/safe/add/AddSafeViewModel.kt [hunk 1]
```diff
@@ -108,9 +111,10 @@ class AddSafeViewModel @Inject constructor(
 
     override fun addAdditionalOwner(input: String): Observable<Result<Unit>> =
             Observable.fromCallable {
-                val address = input.hexAsEthereumAddressOrNull()
```
Comment (rmeissner) [inline app/src/main/java/pm/gnosis/heimdall/ui/safe/add/AddSafeViewModel.kt]: I would still disallow the direct input of something like `ethereum:0x0123...` and only use `parseEthereumAddress` for scanned results (in this case DeployNewSafeFragment). 

Anyways if you leave it in the view model can you add a test for this ;)

File: app/src/main/java/pm/gnosis/heimdall/ui/safe/add/AddSafeViewModel.kt [hunk 2]
```diff
@@ -108,9 +111,10 @@ class AddSafeViewModel @Inject constructor(
...
+                val address = parseEthereumAddress(input)
                 SimpleLocalizedException.assert(address != null, context, R.string.invalid_ethereum_address)
-                SimpleLocalizedException.assert(deviceInfo?.let { it != address } ?: false, context, R.string.error_owner_already_added)
+                SimpleLocalizedException.assert(deviceInfo?.let { it != address }
+                        ?: false, context, R.string.error_owner_already_added)
```
Comment (rmeissner) [inline app/src/main/java/pm/gnosis/heimdall/ui/safe/add/AddSafeViewModel.kt]: that new line is kind a weird :D

Comment (fmrsabino) [reply app/src/main/java/pm/gnosis/heimdall/ui/safe/add/AddSafeViewModel.kt]: Autoformat I think.

Comment (rmeissner) [reply app/src/main/java/pm/gnosis/heimdall/ui/safe/add/AddSafeViewModel.kt]: Yeah, i guessed so, it's not important ;)

File: app/src/main/res/menu/address_book_entry_details_menu.xml [hunk 3]
```diff
@@ -5,10 +5,12 @@
         android:id="@+id/address_book_entry_details_menu_share"
         android:icon="@drawable/ic_share_24dp"
         android:title="@string/share_address"
+        app:iconTint="@color/gnosis_dark_blue_alpha_70"
```
Comment (rmeissner) [inline app/src/main/res/menu/address_book_entry_details_menu.xml]: unnecessary

File: app/src/main/res/menu/address_book_entry_details_menu.xml [hunk 4]
```diff
@@ -5,10 +5,12 @@
...
         app:showAsAction="ifRoom" />
     <item
         android:id="@+id/address_book_entry_details_menu_delete"
         android:icon="@drawable/ic_delete_24dp"
         android:title="@string/delete"
+        app:iconTint="@color/gnosis_dark_blue_alpha_70"
```
Comment (rmeissner) [inline app/src/main/res/menu/address_book_entry_details_menu.xml]: unnecessary

**Pull request merged**